k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -2943,11 +2949,17 @@ public void run() {
      */
     public void sendMessage(final Account account,
                             final Message message,
+                            final MessageReference parentMessageReference,
                             MessagingListener listener) {
         try {
             LocalStore localStore = account.getLocalStore();
             LocalFolder localFolder = localStore.getFolder(account.getOutboxFolderName());
             localFolder.open(Folder.OPEN_MODE_RW);
+            if (account.isBundleMessageFollowings()
+                    && (parentMessageReference != null)
+                    && parentMessageReference.getAccountUuid().equals(account.getUuid())) {
+                outboundSentFolderNamesCache.put(message.getMessageId(), parentMessageReference.getFolderName());
```

Comment:
I don't like this mechanism. The information where a message is moved to after sending should be persisted in the database.
However, I'm undecided on what's the best way to accomplish this. I'm leaning towards a generic solution that allows storing key/value pairs per message. An alternative would be to also use the 'pending commands' queue for sending messages.